tests/baselines/reference/ambientShorthand_reExport.js

```diff
@@ -20,11 +20,19 @@ x($);
 "use strict";
 exports.__esModule = true;

```

Comment:
If you're going with `defineProperty` then the `__esModule` property also probably should be defined (non-enumerable, non-configurable, non-writable); as it is it's all three.

Reply:
It looks like there's some code that already handles using `Object.defineProperty` when the script target is not ES3:

https://github.com/microsoft/TypeScript/blob/27941d05b640148554dd60516144d04a2488536c/src/compiler/transformers/module/module.ts#L1451-L1478